common/include/pcl/pcl_macros.h

```diff
@@ -448,3 +448,34 @@ aligned_free(void* ptr)
 #else
   #define PCL_IF_CONSTEXPR(x) if (x)
 #endif
+
+// [[unlikely]] can be used on any conditional branch, but __builtin_expect is restricted to the evaluation point
+// This makes it quite difficult to create a single macro for switch and while/if
+/**
+ * @def PCL_CONDITION_UNLIKELY
+ * @brief Tries to inform the compiler to optimize codegen assuming that the condition will probably evaluate to false
+ * @note Prefer using `PCL_{IF,WHILE}_UNLIKELY`
+ * @warning This can't be used with switch statements
+ * @details This tries to help the compiler optimize for the unlikely case.
+ * Most compilers assume that the condition would evaluate to true in if and while loops (reference needed)
```

Comment:
> Do you have a reference for this?

[godbolt](godbolt.org/z/11W1z9131) is the best I can do. I don't think this is formalized, just something observed by me in the codegen of several compilers.

I've tried to find some doc for gcc or clang/llvm, but found none so far. Need help here

Reply:
I have no idea where this could be documented, and the godbolt code is at least some exemplary proof, so I am okay with leaving this formulation as it is